src/models.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Class to handle configuration and relation databag."""
+from enum import Enum
+from typing import List, Optional
+
+from charms.data_platform_libs.v0.data_models import RelationDataModel
+from pydantic import BaseModel, ValidationError, validator
+
+
+class AppType(str, Enum):
+    """Class for the app type."""
+
+    PRODUCER = "producer"
+    CONSUMER = "consumer"
+
+
+class CharmConfig(BaseModel):
+    """Charmed configuration class."""
+
+    topic_name: str
+    role: str
```

Comment:
**thought** Initially I had designed this to allow to run the same unit as producer and consumer. Therefore, role was a list and some extra parsing-validating logic was required to go from having a str to a List[AppType]. Now it is not like this anymore, and role can only be 1 value. Therefore, it would make the code easier and cleaner to just have

```
class CharmConfig:
    role: AppType
```

and skip the property and the validation all together. 

PS. If it is easier we could also use `app_type` for the config key. That would probably reduce the number of changes in the code actually. Up to you!